packages/browser/src/instrument.ts

```diff
@@ -0,0 +1,499 @@
+/* tslint:disable:only-arrow-functions no-unsafe-any */
+
+import { API, getCurrentHub } from '@sentry/core';
+import { WrappedFunction } from '@sentry/types';
+import {
+  fill,
+  getFunctionName,
+  getGlobalObject,
+  isString,
+  logger,
+  supportsHistory,
+  supportsNativeFetch,
+} from '@sentry/utils';
+
+import { BrowserClient } from './client';
+
+const global = getGlobalObject<Window>();
+
+/** Object describing handler that will be triggered for a given `type` of instrumentation */
+interface InstrumentHandler {
+  type: InstrumentHandlerType;
+  callback: InstrumentHandlerCallback;
+}
+type InstrumentHandlerType = 'console' | 'dom' | 'fetch' | 'history' | 'sentry' | 'xhr';
+type InstrumentHandlerCallback = (data: any) => void;
+
+/**
+ * Instrument native APIs to call handlers that can be used to create breadcrumbs, APM spans etc.
+ *  - Console API
+ *  - Fetch API
+ *  - XHR API
+ *  - History API
+ *  - DOM API (click/typing)
+ */
+
+const handlers: { [key in InstrumentHandlerType]?: InstrumentHandlerCallback[] } = {};
+const instrumented: { [key in InstrumentHandlerType]?: boolean } = {};
+
+/** Instruments given API */
+function instrument(type: InstrumentHandlerType): void {
+  if (instrumented[type]) {
+    return;
+  }
+
+  instrumented[type] = true;
+
+  switch (type) {
+    case 'console':
+      instrumentConsole();
+      break;
+    case 'dom':
+      instrumentDOM();
+      break;
+    case 'xhr':
+      instrumentXHR();
+      break;
+    case 'fetch':
+      instrumentFetch();
+      break;
+    case 'history':
+      instrumentHistory();
+      break;
+    default:
+      logger.warn('unknown instrumentation type:', type);
+  }
+}
+
+/**
+ * Add handler that will be called when given type of instrumentation triggers
```

Comment:
Let's add `@hidden` here to not have it in TypeDocs.
Also another sentence of that "Use at own risk, this might break without changelog notice, only used internally"